sweep/txgenerator.go

```diff
@@ -169,14 +169,16 @@ func createSweepTx(inputs []input.Input, outputPkScript []byte,
 	}
 
 	// Sweep as much possible, after subtracting txn fees.
-	sweepAmt := int64(totalSum - txFee)
+	sweepAmt := totalSum - txFee
 
 	// The txn will sweep the amount after fees to the pkscript generated
 	// above.
-	sweepTx.AddTxOut(&wire.TxOut{
-		PkScript: outputPkScript,
-		Value:    sweepAmt,
-	})
+	if sweepAmt >= dustLimit {
```

Comment:
Hmmm, IIRC we already filtered this out elsewhere when we compute the "negative yield" outputs? So this may actually be unnecessary.

Reply:
It is also called from the `rpcserver`: `CraftSweepAllTx`, where we don't have this guarantee. And if the resulting output is dust, the non-output TX will fail the transaction sanity check below.